File: src/main/java/com/dec/decisland/network/ClientPayloadHandler.java
package com.dec.decisland.network;

import com.dec.decisland.client.gui.ClientManaOverlay;
import net.minecraft.network.chat.Component;
import net.neoforged.neoforge.network.handling.IPayloadContext;

public class ClientPayloadHandler {
    private static final ClientPayloadHandler INSTANCE = new ClientPayloadHandler();

    public static ClientPayloadHandler getInstance() {
        return INSTANCE;
    }

    public void handleAttack(AttackPayload payload, IPayloadContext context) {
        // 原有的攻击处理逻辑
    }

    // 处理mana同步包
    public void handleManaSync(ManaSyncPayload payload, IPayloadContext context) {
        context.enqueueWork(() -> {
            // 在主线程中更新mana UI
            ClientManaOverlay.updateManaValues(payload.currentMana(), payload.maxMana());
        }).exceptionally(e -> {
            context.disconnect(Component.translatable("decisland.networking.failed", e.getMessage()));
            return null;
        });
    }
}

File: src/main/java/com/dec/decisland/network/ClientPayloadHandler.kt
package com.dec.decisland.network

import com.dec.decisland.client.RecoilClient
import com.dec.decisland.client.bedrock.BedrockEmitterManager
import com.dec.decisland.client.gui.ClientManaOverlay
import net.minecraft.client.Minecraft
import net.minecraft.network.chat.Component
import net.minecraft.world.phys.Vec3
import net.neoforged.neoforge.network.handling.IPayloadContext

object ClientPayloadHandler {
    fun handleAttack(payload: AttackPayload, context: IPayloadContext) {
    }

    fun handleManaSync(payload: ManaSyncPayload, context: IPayloadContext) {
        context.enqueueWork {
            ClientManaOverlay.updateManaValues(payload.currentMana, payload.maxMana)
        }.exceptionally { error ->
            context.disconnect(Component.translatable("decisland.networking.failed", error.message ?: "unknown"))
            null
        }
    }

    fun handleRecoil(payload: RecoilPayload, context: IPayloadContext) {
        context.enqueueWork {
            RecoilClient.kick(payload.pitchUpDegrees, payload.yawDegrees)
        }.exceptionally { error ->
            context.disconnect(Component.translatable("decisland.networking.failed", error.message ?: "unknown"))
            null
        }
    }

    fun handleSpawnBedrockEmitter(payload: SpawnBedrockEmitterPayload, context: IPayloadContext) {
        context.enqueueWork {
            val client = Minecraft.getInstance()
            if (client.level != null) {
                BedrockEmitterManager.spawnAt(payload.id, Vec3(payload.x, payload.y, payload.z), payload.durationTicks)
            }
        }.exceptionally { error ->
            context.disconnect(Component.translatable("decisland.networking.failed", error.message ?: "unknown"))
            null
        }
    }
}
